### PR TITLE
HBASE-28600 Introduce hfile.block.cache.memory.size configuration

### PR DESCRIPTION
**Pull request by jinhyukify**
Introduce a new configuration, `hfile.block.cache.memory.size`, which allows setting the block cache size with an exact byte value or a human-readable storage size.

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 55s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 23s |  |  master passed  |
| +1 :green_heart: |  compile  |   4m 53s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 58s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   2m 27s |  |  master passed  |
| +0 :ok: |  refguide  |   3m 22s |  |  branch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
| +1 :green_heart: |  spotless  |   1m  0s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 47s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   4m 14s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   4m 14s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 56s |  |  the patch passed  |
| +1 :green_heart: |  xmllint  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  spotbugs  |   3m 45s |  |  the patch passed  |
| +0 :ok: |  refguide  |   3m 57s |  |  patch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
| +1 :green_heart: |  hadoopcheck  |  16m 54s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   1m  9s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 25s |  |  The patch does not generate ASF License warnings.  |
|  |   |  63m 51s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6422 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless xmllint refguide |
| uname | Linux 29b3aa620043 5.4.0-195-generic #215-Ubuntu SMP Fri Aug 2 18:28:05 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 05c8c8ac1cd708e12c5d0ab7a81f8f4dadb006af |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| refguide | https://nightlies.apache.org/hbase/HBase-PreCommit-GitHub-PR/PR-6422/1/yetus-general-check/output/branch-site/book.html |
| refguide | https://nightlies.apache.org/hbase/HBase-PreCommit-GitHub-PR/PR-6422/1/yetus-general-check/output/patch-site/book.html |
| Max. process+thread count | 83 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/1/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 xmllint=20913 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 55s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 25s |  |  master passed  |
| +1 :green_heart: |  compile  |   1m 39s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 55s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   6m 34s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 30s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 57s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 57s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m  1s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m 20s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 24s |  |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  | 276m  2s |  |  hbase-server in the patch passed.  |
|  |   | 312m 44s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/1/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6422 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 51254bc775c2 5.4.0-195-generic #215-Ubuntu SMP Fri Aug 2 18:28:05 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 05c8c8ac1cd708e12c5d0ab7a81f8f4dadb006af |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/1/testReport/ |
| Max. process+thread count | 4583 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/1/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Review by ndimiduk — CHANGES_REQUESTED** [review]
This looks really good. I have just some small comments. Have you tried it out? Is it working for you?

File: hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HeapMemoryManager.java [hunk 1]
```diff
@@ -161,14 +161,16 @@ private boolean doInit(Configuration conf) {
     blockCachePercentMaxRange = conf.getFloat(BLOCK_CACHE_SIZE_MAX_RANGE_KEY, blockCachePercent);
     if (blockCachePercent < blockCachePercentMinRange) {
       LOG.warn("Setting " + BLOCK_CACHE_SIZE_MIN_RANGE_KEY + " to " + blockCachePercent
-        + ", same value as " + HFILE_BLOCK_CACHE_SIZE_KEY
+        + " (lookup order: " + HFILE_BLOCK_CACHE_MEMORY_SIZE_KEY + " -> "
```
[the remark this answers is not in the document]
Comment (jinhyukify) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HeapMemoryManager.java:164]: Thank you. updated.

**Review by jinhyukify — COMMENTED** [review]
> Have you tried it out? Is it working for you?

I briefly wrote down the developer test I performed on my local machine.
You can check it in this [comment](https://issues.apache.org/jira/browse/HBASE-28600?focusedCommentId=17903856&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17903856)

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 43s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  7s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 44s |  |  master passed  |
| +1 :green_heart: |  compile  |   3m 14s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 48s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   1m 57s |  |  master passed  |
| +0 :ok: |  refguide  |   2m  6s |  |  branch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
| +1 :green_heart: |  spotless  |   0m 41s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 40s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 12s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 12s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 48s |  |  the patch passed  |
| +1 :green_heart: |  xmllint  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  spotbugs  |   2m  9s |  |  the patch passed  |
| +0 :ok: |  refguide  |   1m 53s |  |  patch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
| +1 :green_heart: |  hadoopcheck  |   9m 21s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| -1 :x: |  spotless  |   0m 34s |  |  patch has 61 errors when running spotless:check, run spotless:apply to fix.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 21s |  |  The patch does not generate ASF License warnings.  |
|  |   |  39m 23s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6422 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless xmllint refguide |
| uname | Linux 4153ea8fd962 5.4.0-200-generic #220-Ubuntu SMP Fri Sep 27 13:19:16 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / d5d85dff52399f3c6d0f0da1f1dc705e01850ad4 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| refguide | https://nightlies.apache.org/hbase/HBase-PreCommit-GitHub-PR/PR-6422/2/yetus-general-check/output/branch-site/book.html |
| refguide | https://nightlies.apache.org/hbase/HBase-PreCommit-GitHub-PR/PR-6422/2/yetus-general-check/output/patch-site/book.html |
| spotless | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/2/artifact/yetus-general-check/output/patch-spotless.txt |
| Max. process+thread count | 84 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/2/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 xmllint=20913 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 35s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m  4s |  |  master passed  |
| +1 :green_heart: |  compile  |   1m 12s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 43s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 47s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 58s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 13s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 13s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 43s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 42s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 11s |  |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  | 222m 24s |  |  hbase-server in the patch passed.  |
|  |   | 251m 48s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/2/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6422 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux fef0d68eb0ac 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / d5d85dff52399f3c6d0f0da1f1dc705e01850ad4 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/2/testReport/ |
| Max. process+thread count | 5147 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/2/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 47s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m  4s |  |  master passed  |
| +1 :green_heart: |  compile  |   1m 12s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 50s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 46s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 40s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 12s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 12s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 44s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m  2s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m  6s |  |  hbase-common in the patch passed.  |
| -1 :x: |  unit  | 228m 44s | [/patch-unit-hbase-server.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/3/artifact/yetus-jdk17-hadoop3-check/output/patch-unit-hbase-server.txt) |  hbase-server in the patch failed.  |
|  |   | 257m  8s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/3/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6422 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 88749b5bfba4 5.4.0-200-generic #220-Ubuntu SMP Fri Sep 27 13:19:16 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / a4015606616a86a9ee89994d313350170b390524 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/3/testReport/ |
| Max. process+thread count | 4632 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/3/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/io/util/MemorySizeUtil.java [hunk 2]
```diff
@@ -207,27 +228,27 @@ public static float getBlockCacheHeapPercent(final Configuration conf) {
...
       throw new IllegalArgumentException(
         HConstants.HFILE_BLOCK_CACHE_SIZE_KEY + " must be between 0.0 and 1.0, and not > 1.0");
     }
-    long max = -1L;
+
     final MemoryUsage usage = safeGetHeapMemoryUsage();
-    if (usage != null) {
-      max = usage.getMax();
-    }
+    final long heapMax = usage != null ? usage.getMax() : -1;
```
[the remark this answers is not in the document]
Comment (jinhyukify) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/io/util/MemorySizeUtil.java]: What do you think of immediately returning `-1` in this case?
We expect this method to return a negative value when the cache is disabled.
https://github.com/apache/hbase/blob/1880cb3ab57f74ada8fd0161a400676a15b4012c/hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/BlockCacheFactory.java#L117-L120

**Review by ndimiduk — APPROVED** [review]
Nice thank you!

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 31s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 12s |  |  master passed  |
| +1 :green_heart: |  compile  |   3m 46s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 49s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   2m  2s |  |  master passed  |
| +0 :ok: |  refguide  |   2m 20s |  |  branch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
| +1 :green_heart: |  spotless  |   0m 46s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m  3s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 46s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 46s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 48s |  |  the patch passed  |
| +1 :green_heart: |  xmllint  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  spotbugs  |   2m 21s |  |  the patch passed  |
| +0 :ok: |  refguide  |   2m  4s |  |  patch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
| +1 :green_heart: |  hadoopcheck  |  11m 28s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 17s |  |  The patch does not generate ASF License warnings.  |
|  |   |  45m 39s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6422 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless xmllint refguide |
| uname | Linux c194f512c438 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / a4015606616a86a9ee89994d313350170b390524 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| refguide | https://nightlies.apache.org/hbase/HBase-PreCommit-GitHub-PR/PR-6422/4/yetus-general-check/output/branch-site/book.html |
| refguide | https://nightlies.apache.org/hbase/HBase-PreCommit-GitHub-PR/PR-6422/4/yetus-general-check/output/patch-site/book.html |
| Max. process+thread count | 86 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/4/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 xmllint=20913 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 43s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 43s |  |  master passed  |
| +1 :green_heart: |  compile  |   1m 17s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 45s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m  0s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m  8s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 17s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 17s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 44s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 51s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m  6s |  |  hbase-common in the patch passed.  |
| -1 :x: |  unit  |  26m 42s | [/patch-unit-hbase-server.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/4/artifact/yetus-jdk17-hadoop3-check/output/patch-unit-hbase-server.txt) |  hbase-server in the patch failed.  |
|  |   |  52m 17s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/4/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6422 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 09253916f505 5.4.0-200-generic #220-Ubuntu SMP Fri Sep 27 13:19:16 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / a4015606616a86a9ee89994d313350170b390524 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/4/testReport/ |
| Max. process+thread count | 2036 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6422/4/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by ndimiduk** [comment]
I think that this change is benign enough for our active release lines. @apurtell do you have any objection to adding this to branch-2.5?

**Pull request merged**